x/twap/api_test.go

```diff
@@ -182,9 +243,98 @@ func (s *TestSuite) TestGetArithmeticTwap() {
 				s.twapkeeper.StoreNewRecord(s.Ctx, record)
 			}
 			s.Ctx = s.Ctx.WithBlockTime(test.ctxTime)
-			twap, err := s.twapkeeper.GetArithmeticTwap(s.Ctx, test.input.poolId,
+
+			var twap sdk.Dec
+			var err error
+
+			twap, err = s.twapkeeper.GetArithmeticTwap(s.Ctx, test.input.poolId,
 				test.input.quoteAssetDenom, test.input.baseAssetDenom,
 				test.input.startTime, test.input.endTime)
+
+			if test.expErrorStr != "" {
+				s.Require().Error(err)
+				s.Require().Contains(err.Error(), test.expErrorStr)
+				return
+			}
+			s.Require().NoError(err)
+			s.Require().Equal(test.expTwap, twap)
+		})
+	}
+}
+
+// TestGetArithmeticTwapToNow tests if we get the expected twap value from `GetArithmeticTwapToNow`.
+// We test the method directly by updating the accumulator and storing the twap records
+// manually in this test.
+func (s *TestSuite) TestGetArithmeticTwapToNow() {
+
+	makeSimpleTwapToNowInput := func(startTime time.Time, isQuoteTokenA bool) getTwapInput {
+		quoteAssetDenom, baseAssetDenom := denom0, denom1
+		if isQuoteTokenA {
+			baseAssetDenom, quoteAssetDenom = quoteAssetDenom, baseAssetDenom
+		}
+		return getTwapInput{1, quoteAssetDenom, baseAssetDenom, startTime, startTime}
+	}
+
+	quoteAssetA := true
+
+	tests := map[string]struct {
+		recordsToSet []types.TwapRecord
+		ctxTime      time.Time
+		input        getTwapInput
+		expTwap      sdk.Dec
+		expErrorStr  string
+	}{
+		"(1 record) start time = record time": {
+			recordsToSet: []types.TwapRecord{baseRecord},
+			ctxTime:      baseTime.Add(time.Minute),
+			input:        makeSimpleTwapToNowInput(baseTime, quoteAssetA),
+			expTwap:      sdk.NewDec(10),
+		},
+		"(1 record) to_now: start time > record time": {
+			recordsToSet: []types.TwapRecord{baseRecord},
+			ctxTime:      baseTime.Add(time.Minute),
+			input:        makeSimpleTwapToNowInput(baseTime.Add(10*time.Second), quoteAssetA),
+			expTwap:      sdk.NewDec(10),
+		},
+		"(2 record) to now: start time = second record time": {
+			recordsToSet: []types.TwapRecord{baseRecord, tPlus10sp5Record},
+			ctxTime:      baseTime.Add(time.Minute),
+			input:        makeSimpleTwapToNowInput(baseTime.Add(10*time.Second), quoteAssetA),
+			expTwap:      sdk.NewDec(5), // .1 for 10s, .2 for 10s
```

Comment:
Still learning the calculation but would this be 
```suggestion
			expTwap:      sdk.NewDec(5), // 10 for 0s, 5 for 10s
```

and here's the calculation ` // (10 * (10s - 10s) + 5 * (20s - 10s)) / (20s - 10s)`

Reply:
correct! Good catch!

Reply:
Hi @mattverse just want to verify this again, is this calculation correct `// (10 * (10s - 10s) + 5 * (20s - 10s)) / (20s - 10s)`, because i'm trying to add 3rd record using the same logic and its a bit off

Reply:
it should be 250 / 50. (350000000000000000000000000000 - 100000000000000000000000000000) / 50 (60-10)